R/bundleFiles.R

```diff
@@ -179,7 +179,7 @@ ignoreBundleFiles <- function(dir, contents) {
     # rsconnect packages
     "rsconnect", "rsconnect-python", "manifest.json",
     # packrat + renv,
-    "renv", "renv.lock", "packrat",
+    "renv", "packrat",
```

Comment:
Yes; that's necessary so that we can process it in `bundlePackages()`. Do you foresee a problem with including this in the deployed bundle? If so, I can delete. (But I think you could also `renv::restore()` directly from that file without having to convert the manifest back into a packrat spec).

Reply:
I am torn on this one.

On the one hand, including only the `manifest.json` in the bundle is quite nice because it means we have a single source-of-truth. Folks have been confused by the presence of both `manifest.json` and `packrat.lock` in our bundles, as well as the relationship between the two (and duplication of information).

On the other hand, I like including the `renv.lock`, as it gives us a future option of directly using that file without reconstructing it from the manifest.

Mostly, my struggle is that I had thought about this change as "manifest-only, no packrat.lock" rather than "manifest plus renv.lock; except when we create the renv.lock on the fly".

Let's go forward with the (sometimes) inclusion, but probably document the change.

Reply:
A generated lock file isn't on the set of "bundle files" that was computed much earlier in the workflow... I'm OK either way, but if we do include a generated lock file, let's include it only when we have tracked dependencies (e.g. not for static).